src/ai/ai.cpp

```diff
@@ -22,23 +57,14 @@ void update_ai_general_status(sys::state& state) {
 		for(auto b : state.world.nation_get_nation_adjacency_as_connected_nations(n)) {
 			auto other = b.get_connected_nations(0) != n ? b.get_connected_nations(0) : b.get_connected_nations(1);
 			if(!nations::are_allied(state, n, other) && (!in_sphere_of || in_sphere_of != other.get_in_sphere_of())) {
-				greatest_neighbor = std::max(greatest_neighbor, float(state.world.nation_get_military_score(other)));
+				greatest_neighbor = std::max(greatest_neighbor, estimate_strength(state, other));
 			}
 		}
 
 		float self_str = float(state.world.nation_get_military_score(n));
-		for(auto subj : n.get_overlord_as_ruler()) {
-			self_str += 0.75f * float(subj.get_subject().get_military_score());
-		}
-
-		float defensive_str = self_str;
-		defensive_str += in_sphere_of ? float(state.world.nation_get_military_score(in_sphere_of)) : 0.0f;
-		for(auto d : state.world.nation_get_diplomatic_relation(n)) {
-			if(d.get_are_allied()) {
-				auto other = d.get_related_nations(0) != n ? d.get_related_nations(0) : d.get_related_nations(1);
-				defensive_str += float(state.world.nation_get_military_score(other));
-			}
-		}
+		for(auto subj : n.get_overlord_as_ruler())
+			self_str += 0.5f * float(subj.get_subject().get_military_score());
```

Comment:
why have you decided to make this 0.5 instead of 0.75?

Reply:
puppets dont help a lot anyways

Reply:
Ok, but see below: since we don't know how things are working yet, changing constants like this is change for the sake of change